### DISCUSSION TITLE
Importing VueObserveVisibility in boot file

### DISCUSSION THREAD

**Discussion by pasjonatprogramowania** [quasarframework/quasar#9844] [General - Components / Directives / etc]
I need in my project to import VueObserveVisibility library but installation example don't work in my quasar project, Can explain to me how can I import this correctly ?

https://github.com/Akryum/vue-observe-visibility#installation

```
import VueObserveVisibility from 'vue-observe-visibility'
export default ( { app, router, Vue  } ) => {
    
    Vue.use(VueObserveVisibility)
}
 ```

![Code_IxaERLsSR5](https://user-images.githubusercontent.com/44599809/123629504-a4289280-d814-11eb-87d8-1068b041f2ad.png)

**Comment by hawkeye64** [comment]
@pasjonatprogramowania I am not sure what the issue is as there is not enough information. That being said, why add extra bloat to your app when Quasar has you covered with similar functionality?
Try the Intersection directive.
https://quasar.dev/vue-directives/intersection#introduction

Or the Intersection component.
https://quasar.dev/vue-components/intersection

You can see the latter being used here: https://github.com/hawkeye64/quasar-svg-icons/blob/master/src/pages/Index.vue#L13-L23

**Reply by pasjonatprogramowania** [reply]
I didn't know abut this feature thank you so much
I'm trying to make infinity scrolling page with cards from JSON data, but rendering all in ones take too much time. From this reason I'm trying only render cards that are visible to user 

like this

https://youtu.be/ZkhgRtbkM8U?t=784

I know about build in quasar `infinity scrolling` component but this use `setup` to render data and in my case this is not a good option 

https://v1.quasar.dev/vue-components/infinite-scroll#usage 

Is this `Intersection` is a possible way to solve this problem ?
https://quasar.dev/vue-directives/intersection#introduction

I am adding my code, this is simple `v-for` that create that cards

```
<template>
            <div>
                  <q-card class="my-card" v-for="(p,k) in returnProductsBySelectedType()" :key="k" flat bordered >
                    <div>
                      <q-item>
                        <q-item-section >
                          <div class="text-body1 q-mt-sm q-mb-xs" >{{ p.name }}</div>
                        </q-item-section>
                        <q-btn flat class="bg-primary" style="width:50px; height: 50px;" color="white"></q-btn>
                      </q-item>
                      <q-separator />
                      <q-item-section class="q-pa-sm bg-grey-3">
                        <div class="text-subtitle2 flex ">Kalorie: (Kcal) Tłuszcze: (g) Weglowodany: (g) Bialko: (g)</div>
                      </q-item-section>
                    </div>
                  </q-card>
                </div>
</template>
<script>
import json from 'data.json'
data:{
    products:json,
}
methods:{
 returnProductsBySelectedType(){
      let arr=[]
      for(const {type,products} of this.products){ 
        if(this.dropdownType === type ) return products
      }
      return arr
}
</script>
```

json
```
[
   {
      "type":"zboża",
      "products":[
         {
            "id":1,
            "name":"Bagietki francuskie"
         },
         {
            "id":2,
            "name":"Bułeczki do hot-dogów"
         },
         {
            "id":3,
            "name":"Bułka tarta"
         },
         {
            "id":4,
            "name":"Bułki grahamki"
         },
         {
            "id":5,
            "name":"Bułki i rogale maślane"
         },
         {
            "id":6,
            "name":"Bułki mieszane, z cebulą"
         },
         {
            "id":7,
            "name":"Bułki mleczne"
         },
         {
            "id":8,
            "name":"Bułki owsiane"
         },
         {
            "id":9,
            "name":"Bułki pszenne zwykłe"
         },
         {
            "id":10,
            "name":"Bułki pszenne zwykłe, z serwatką"
         },
         {
            "id":11,
            "name":"Bułki sojowe"
         },
         {
            "id":12,
            "name":"Bułki szwedki"
         },
         {
            "id":13,
            "name":"Bułki wrocławskie"
         },
         {
            "id":14,
            "name":"Chałki zdobne"
         },
         {
            "id":15,
            "name":"Chleb baltonowski"
         },
         {
            "id":16,
            "name":"Chleb beskidzki"
         },
         {
            "id":17,
            "name":"Chleb chrupki"
         },
      ]
   }
]
```

![nfx32](https://user-images.githubusercontent.com/44599809/123670138-2e84ec80-d83d-11eb-8fb4-628ccdc56d35.png)

**Reply by metalsadman** [reply]
you're adding your directive in the closing tag. `</div v-....>` seem wrong.

**Reply by pasjonatprogramowania** [reply]
This is code from external library for vue https://github.com/Akryum/vue-observe-visibility#installation
